plutus-use-cases/src/Language/Plutus/Tutorial.lhs

```diff
@@ -0,0 +1,159 @@
+> module Language.Plutus.Tutorial where
+
+This tutorial shows how to implement a simple crowdfunding campaign as a Plutus contract. We need some imports:
+
+> import           GHC.Generics (Generic)
+> import qualified Language.PlutusTx          as PlutusTx
+> import qualified Language.Plutus.Runtime    as Runtime
+> import qualified Language.Plutus.Runtime.TH as Runtime
+> import qualified Wallet.UTXO                as UTXO
+> import           Wallet.UTXO                (Validator (..))
+> import           Prelude                    hiding ((&&))
+
+The campaign has the following parameters:
+
+* Funding target
+* End date
+* Collection deadline
+* Campaign owner
+
+If the funding target is reached at the end date, then the campaign owner may collect all the funds. If it isn't reached, or if the owner does not collect the funds before the collection deadline, then the contributors are entitled to a refund.
+
+In Haskell:
+
+> data Campaign = Campaign {
+>       fundingTarget      :: Runtime.Value,
+>       endDate            :: Runtime.Height,
+>       collectionDeadline :: Runtime.Height,
+>       campaignOwner      :: Runtime.PubKey
+>  } deriving (Generic)
+
+The type of monetary values is `Runtime.Value`. Dates are expressed in terms of blocks, and they type is `Runtime.Height`. The campaign owner is identified by their public key.
+
+One of the strengths of Plutus is the ability to use the same definitions for on-chain and off-chain code, which includes lifting values from Haskell to Plutus Core. To enable values of the `Campaign` type to be lifted, we need to call `makeLift` from the `PlutusTx` module (which in turn requires the `deriving (Generic)` clause in the data type):
+
+> PlutusTx.makeLift ''Campaign
+
+Now we need to figure out what the campaign will look like on the blockchain. Which transactions are involved, who submits them, and in what order? 
+
+Each contributor pays their contribution to the address of the campaign script. When the block height `endDate` is reached, the campaign owner submits a single transaction that consuming all inputs from the campaign address and paying them to a pubkey address. If the funding target isn't reached, or the campaign owner fails to collect the funds, then each contributor can claim a refund, in the form of a transaction that consumes their own contribution. This means that the validator script is going to be run once per contribution, and we need to tell it which of the two cases outcomes it should check.
+
+We can encode the two possible actions in a data type:
+
+> data CampaignAction = Collect | Refund
+>   deriving (Generic)
+> PlutusTx.makeLift ''CampaignAction
+
+The `CampaignAction` will be submitted as the redeemer script. Now we need one final bit of information, namely the identity (public key) of each contributor, so that we know the recipient of the refund. This public key is going to be stored in the data script of the contribution. 
```

Comment:
Reading this I immediately wonder why we don't just put the pair of the action and the key in the redeemer script.

Reply:
To ensure that only the contributor can make a reclaim (I added a comment in the text)